x-pack/plugins/log_explorer/public/components/flyout_detail/flyout_highlights.tsx

```diff
@@ -47,156 +49,164 @@ export function FlyoutHighlights({
 }) {
   const [ref, dimensions] = useMeasure<HTMLDivElement>();
   const { columns, fieldWidth } = useFlyoutColumnWidth(dimensions.width);
+
   return (
     <DiscoverActionsProvider value={actions}>
       <HighlightContainer ref={ref}>
+        {/* Service highlight */}
         <HighlightSection title={serviceAccordionTitle} columns={columns}>
           {formattedDoc[constants.SERVICE_NAME_FIELD] && (
             <HighlightField
-              label={flyoutServiceLabel}
+              data-test-subj="logExplorerFlyoutService"
               field={constants.SERVICE_NAME_FIELD}
-              value={flattenedDoc[constants.SERVICE_NAME_FIELD]}
               formattedValue={formattedDoc[constants.SERVICE_NAME_FIELD]}
-              dataTestSubj="logExplorerFlyoutService"
+              icon={<AgentIcon agentName={formattedDoc[constants.AGENT_NAME_FIELD] as AgentName} />}
```

Comment:
Can you try to add a filter for Agent Name and see how this value is returned ?
i suspect since we are retrieving it from `formattedDoc` it will come back as `<mark>AgentName</mark>

Reply:
Good catch! It would happen what you said, the problem with the flattened doc is that the value is wrapped in an array, so I'll have to extract it. On it.